File: java/bob/src/main/java/Bob.java
public class Bob {
    public String hey(String input) {
        String cleanedInput = input.trim();
        Boolean question = cleanedInput.endsWith("?");
        Boolean yell = cleanedInput.chars().anyMatch(Character::isUpperCase) &&
                       cleanedInput.chars().noneMatch(Character::isLowerCase);
        if (question && yell) {
            return "Calm down, I know what I'm doing!";
        } else if (question) {
            return "Sure.";
        } else if (yell) {
            return "Whoa, chill out!";
        } else if (cleanedInput.chars().allMatch(Character::isWhitespace)) {
            return "Fine. Be that way!";
        } else {
            return "Whatever.";
        }
    }
}

File: kotlin/bob/src/main/kotlin/Bob.kt
object Bob {
    fun hey(input: String): String = when {
        input.trim().endsWith('?') -> if (yelling(input)) "Calm down, I know what I'm doing!" else "Sure."
        silence(input) -> "Fine. Be that way!"
        else -> if (yelling(input)) "Whoa, chill out!" else "Whatever."
    }

    private fun yelling(input: String) = input.any { it.isUpperCase() } && input.none { it.isLowerCase() }

    private fun silence(input: String) = input.all { it.isWhitespace() }
}
